subsys/bluetooth/host/smp.c

```diff
@@ -3995,7 +4029,10 @@ static uint8_t smp_ident_addr_info(struct bt_smp *smp, struct net_buf *buf)
 			}
 		}
 
-		bt_id_add(keys);
+		err = smp_id_add_unpair(smp, keys);
```

Comment:
should you also change https://github.com/zephyrproject-rtos/zephyr/blob/a4c046cd5a89a513483d164a17ec0e3178be07dc/subsys/bluetooth/host/smp.c#L4786 ?

Reply:
I think yes. At least it can't hurt. I'll see if I can make a bablesim test that fails there.